server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -92,88 +67,27 @@ protected ObjectMapper createMapper(String name, Integer position, String fullPa
-                                       Version indexVersionCreated) {
-            if (fieldName.equals("date_formats") || fieldName.equals("dynamic_date_formats")) {
-                if (fieldNode instanceof List) {
-                    List<FormatDateTimeFormatter> formatters = new ArrayList<>();
-                    for (Object formatter : (List<?>) fieldNode) {
-                        if (formatter.toString().startsWith("epoch_")) {
-                            throw new MapperParsingException(
-                                "Epoch [" + formatter + "] is not supported as dynamic date format");
-                        }
-                        formatters.add(parseDateTimeFormatter(formatter));
-                    }
-                    builder.dynamicDateTimeFormatter(formatters);
-                } else if ("none".equals(fieldNode.toString())) {
-                    builder.dynamicDateTimeFormatter(Collections.emptyList());
-                } else {
-                    builder.dynamicDateTimeFormatter(Collections.singleton(parseDateTimeFormatter(fieldNode)));
-                }
-                return true;
-            }
-            return false;
-        }
     }
 
-    private Explicit<FormatDateTimeFormatter[]> dynamicDateTimeFormatters;
-
     RootObjectMapper(String name,
                      Dynamic dynamic,
                      Map<String, Mapper> mappers,
-                     Explicit<FormatDateTimeFormatter[]> dynamicDateTimeFormatters,
                      Settings settings) {
         super(name, null, name, dynamic, mappers, settings);
-        this.dynamicDateTimeFormatters = dynamicDateTimeFormatters;
-    }
-
-    @Override
-    public ObjectMapper mappingUpdate(Mapper mapper) {
-        RootObjectMapper update = (RootObjectMapper) super.mappingUpdate(mapper);
-        // for dynamic updates, no need to carry root-specific options, we just
-        // set everything to they implicit default value so that they are not
-        // applied at merge time
-        update.dynamicDateTimeFormatters = new Explicit<FormatDateTimeFormatter[]>(Defaults.DYNAMIC_DATE_TIME_FORMATTERS, false);
-        return update;
-    }
-
-    public FormatDateTimeFormatter[] dynamicDateTimeFormatters() {
-        return dynamicDateTimeFormatters.value();
     }
 
     @Override
     public RootObjectMapper merge(Mapper mergeWith) {
         return (RootObjectMapper) super.merge(mergeWith);
     }
 
-    @Override
-    protected void doMerge(ObjectMapper mergeWith) {
-        super.doMerge(mergeWith);
-        RootObjectMapper mergeWithObject = (RootObjectMapper) mergeWith;
-        if (mergeWithObject.dynamicDateTimeFormatters.explicit()) {
-            this.dynamicDateTimeFormatters = mergeWithObject.dynamicDateTimeFormatters;
-        }
-    }
-
     @Override
     protected void doXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
```

Comment:
Looks like this method is not used anymore now, so we could remove it from `ObjectMapper`?

Reply:
Right, good catch. Pushed a fixup